### PR TITLE
fix: [CDS-45858]: Prevent reading logs from tf output and limit error message size

### PR DESCRIPTION
**Pull request by abosii-harness**
## Describe your changes
When TF_LOG is enabled with any level terraform cli will put all logs in error stream. In scenario where in all these logs were'n able to identify a meaningful for us error it will result in just pushing all these logs in error message. This PR is solving 2 issues:
- Filter all logs related line from error stream in cli error output
- Limit possible error message size for a particular exception (Note: final error chain can be composed by multiple error messages)

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/39519)

### CODE REVIEW

**Review by tmacari-harness — APPROVED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39519)* status: :shipit: complete! ✓

File: 960-api-services/src/main/java/io/harness/terraform/TerraformClientImpl.java [hunk 1]
```diff
@@ -55,6 +56,8 @@
 @Singleton
 @OwnedBy(CDP)
 public class TerraformClientImpl implements TerraformClient {
+  private static final Pattern TF_LOG_ERROR_STREAM_PATTERN =
```
Comment (vlica-harness) [inline 960-api-services/src/main/java/io/harness/terraform/TerraformClientImpl.java:59]: minor,  not sure i got it right, but naming (_ERROR_) in TF_LOG_ERROR_STREAM_PATTERN includes INFO,DEBUG. is this expected naming?

**Review by vlica-harness — DISMISSED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39519)* status: :shipit: complete! ✓

**Review by vlica-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39519)* status: :shipit: complete! ✓

File: 960-api-services/src/main/java/io/harness/terraform/TerraformClientImpl.java (hunk 1 above)
Comment (abosii-harness) [reply 960-api-services/src/main/java/io/harness/terraform/TerraformClientImpl.java]: Renamed

**Review by tmacari-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39519)* status: 1. unresolved discussions.

**Review by vlica-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39519)* status: :shipit: complete! ✓

**Pull request merged**